### PR TITLE
Fix organization slug computation clashes

### PR DESCRIPTION
**Pull request by sharat87**

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/CustomOrganizationRepositoryImpl.java [hunk 1]
```diff
@@ -39,4 +42,28 @@ public CustomOrganizationRepositoryImpl(ReactiveMongoOperations mongoOperations,
...
+        return mongoOperations
+                .find(slugPrefixQuery, Organization.class)
+                .map(Organization::getSlug)
+                .collect(Collectors.toSet())
+                .map(slugs -> {
+                    if (slugs.isEmpty()) {
+                        return 0L;
+                    }
+
+                    long number = 1L;
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/CustomOrganizationRepositoryImpl.java:60]: There's one tine edge case of slugs never getting their own names (without numbers) if there are already numbered slugs present. But this takes car of unique slugs at least, so we can push this in if you think the edge case doesn't matter

**Review by nidhi-nair — DISMISSED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/CustomOrganizationRepositoryImpl.java (hunk 1 above)
Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/CustomOrganizationRepositoryImpl.java:60]: Actually, number-less slugs are indeed used. The `0` returned here is handled as such [up the stack](https://github.com/appsmithorg/appsmith/blob/c129922c9035832ff515fc8b6432851e8679c57d/app/server/appsmith-server/src/main/java/com/appsmith/server/services/OrganizationServiceImpl.java#L102).

Also, we got [tests to ensure this](https://github.com/appsmithorg/appsmith/blob/c129922c9035832ff515fc8b6432851e8679c57d/app/server/appsmith-server/src/test/java/com/appsmith/server/services/OrganizationServiceTest.java#L269) :)

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/CustomOrganizationRepositoryImpl.java:60]: Wait, sorry, you're right. I misunderstood. Let me see if I can handle it.

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/CustomOrganizationRepositoryImpl.java:60]: Excellent eye! Fixed in another commit. Can you review again please? Thanks!

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**